### DISCUSSION TITLE
All compilation commands skipped

### DISCUSSION THREAD

**Discussion by d0rata** [Ericsson/codechecker#3906] [Q&A] [answered]
I'm trying to work through the quick start [guide](https://codechecker.readthedocs.io/en/latest/usage/#step-1) using the example code in the codechecker repo, but am running into issues when trying to run `CodeChecker analyze`. The `log` step appears to have worked, because my compilation database is not empty and the `ccexample` binary is built. However, when I attempt the `analyze` step, I get the following output:

```
$CodeChecker analyze ./compile_commands.json --enable sensitive --output ./reports
[INFO 2023-05-11 00:27] - No analysis is required.
There were no compilation commands in the provided compilation database or all of them were skipped.
```

Any thoughts on what the problem could be?

**Comment by bruntib** [comment]
Hi, could you attach the `compile_commands.json` file, please?
Also, do you have a clang/clang-tidy/cppcheck installed on your machine?

**Comment by d0rata** [comment]
```
$ clang --version
Ubuntu clang version 14.0.0-1ubuntu1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

```
$ clang-tidy --version
Ubuntu LLVM version 14.0.0
Optimized build.
Default target: x86_64-pc-linux-gnu
Host CPU: znver2
```

```
$cppcheck --version
Cppcheck 2.7
```
[compile_commands.json.txt](https://github.com/Ericsson/codechecker/files/11454866/compile_commands.json.txt)

**Reply by bruntib** [reply, marked as answer]
This `compile_commands.json.txt` contains compiler invocations that do both preprocessing and compilation. CodeCompass skips the build actions that are related to preprocessing. For example the usage of `-MP` flag indicates that these build actions are controlling some preprocessing actions.

However, the core problem is that you're probably using `make` >= 4.3 version during `CodeChecker log` command. This contains such a setting by default (see the release notes https://lists.gnu.org/archive/html/info-gnu/2020-01/msg00004.html):
```
* GNU make will now use posix_spawn() on systems where it is available.
  If you prefer to use fork/exec even on systems where posix_spawn() is
  present, you can use the --disable-posix-spawn option to configure.
  Implementation contributed by Aron Barath <address@hidden>
```
CodeChecker logger is not supporting the logging of these system calls, so that's why only the subprocesses are logged that are started as a subprocess by `gcc` compiler. For example `cc1` can be seen at the beginning of the build commands in the attached `compile_commands.json.txt` file.

So the solution is either using `make` < 4.3 while logging with CodeCompass, or rebuild `make` tool itself with `--disable-posix-spawn` flag from source as indicated in the documentation above.
In the long terms we should improve CodeChecker logger so `posix_spawn()` calls are also captured.

**Reply by d0rata** [reply]
Yes, this was the issue. Thank you for the guidance.
